pages/admin/lessons/[lessonSlug]/[pageName]/index.tsx

```diff
@@ -193,7 +193,9 @@ const ExercisesPage = ({ lessonSlug }: ExercisesProps) => {
   const mapExercisesToExerciseCard = data?.exercises
     .filter(
       exercise =>
-        exercise.flaggedAt && exercise.module.lesson.slug === lessonSlug
+        exercise.flaggedAt &&
+        !exercise.removedAt &&
```

Comment:
I added a condition to prevent the display of any removed exercises. Sending removed exercises to the client would be a waste of bandwidth, as they will not be displayed. Instead, it would be more efficient to have the backend filter out removed exercises and only send the ones that should be displayed to the client.